sematic/resolvers/tests/test_local_resolver.py

```diff
@@ -382,6 +458,20 @@ def pipeline():
     assert runs_by_id[future.nested_future.id].future_state == FutureState.FAILED.value
     assert "FAIL!" in runs_by_id[future.nested_future.id].exception_metadata.repr
 
+    parent_callbacks = resolver.callback_by_future_id(future.id)
+    nested_callbacks = resolver.callback_by_future_id(future.nested_future.id)
+    assert parent_callbacks == [
+        "_future_did_schedule",
+        "_future_did_run",
+        "_future_did_fail",
+        "_future_did_terminate",
+    ]
+    assert nested_callbacks == [
+        "_future_did_schedule",
+        "_future_did_fail",
+        "_future_did_terminate",
+    ]
```

Comment:
These checks don't actually validate the `LocalResolver`, but the external resource feature. Also, they supplant the `LocalResolver` with a different one, and test that.

Please extract `CallbackTrackingResolver` and all these new checks and test cases to `test_external_resource.py`.